### DISCUSSION TITLE
Block Reservations API for Admin

### DISCUSSION THREAD

**Discussion by amitjattan** [Azure/ipam#88] [Ideas]
Hey @DCMattyG This is Amit from SIC. Can you please check this feature and let me know if this is relevant ask? We have been asked from our product owners that we should have the reservation APIs blocked only accessible for Admins. Because if we rollout this API to our end development teams they should create DOS attacks and fill all the block reservations. There should be a way to restrict the users when they request a vnet creation they should be able to create vnets only based on approved reservations which Admins of the tool will allow them to use.

**Comment by DCMattyG** [comment]
Hey there @amitjattan, wonderful to hear from you!

The first thing I'll say here is that Azure IPAM was designed around providing a way of automating IP Address Management through API's. A typical pattern here is for an automated process/pipeline to call out to Azure IPAM for a reservation, retrieve the CIDR that is returned, then apply that CIDR to a new Virtual Network in Azure. I'm curious how this would look in your scenario above...

If you are pre-approving reservations (CIDR's), why not just provide the end user the CIDR directly instead of going though automation?

Can you help me understand a bit more about how this workflow would look from a user/consumers perspective? Would they email or put in a ticket for a CIDR (through something like Service Now), wait for the approval, then that would put the reservation into Azure IPAM, then once all of that happened they could run their automation to build the new Virtual Network?

**Reply by amitjattan** [reply]
So we already have a centralized Pipeline for Landing Zones automations where original request is fulfilling the creation of Subscriptions, RBACs. VNETs etc. And if they want to create more vNets they just update their Landing Zones data templates and it automatically creates the vNets for them base don what size they preferred. But our concern is how to avoid end clients to not use the reservations if we rollout, but they should still be able to use this for subnets CIDR identifications.
